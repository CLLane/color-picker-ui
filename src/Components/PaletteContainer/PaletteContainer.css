.palette-card__container {
  display: flex;
   width: 100px;
  height: 300px;
  overflow-y: auto;
  overflow-x: hidden;
}