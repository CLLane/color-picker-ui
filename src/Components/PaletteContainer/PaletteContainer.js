import React from "react";
import "./PaletteContainer.css";
import PaletteCard from "../PaletteCard/PaletteCard";

export const PaletteContainer = ({ palettes }) => {
  const paletteCards = palettes.map(palette => {
    return <PaletteCard palette={palette} key={palette.id} />;
  });

  return <section>{paletteCards}</section>;
};

export default PaletteContainer;
