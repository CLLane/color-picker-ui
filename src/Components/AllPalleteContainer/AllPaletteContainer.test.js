import React from 'react';
import { shallow } from 'enzyme';
import './PaletteContainer.css';
import PaletteContainer from './PaletteContainer';

describe('Palette Container', () => {
  let wrapper;
  let mockPalettes;
  let mockGrabPalette;
  let mockUpdatePaletteName;
  beforeEach(() => {
    mockPalettes = [
      {
        id: 1, 
        name: 'Cool Colors',
        project_id: '12345',
        colorOne: '#FFFFFF',
        colorTwo: '#FFFFFF',
        colorThree: '#FFFFFF',
        colorFour: '#FFFFFF',
        colorFive: '#FFFFFF'
      },
      {
        id: 2, 
        name: 'Warm Colors',
        project_id: '12345',
        colorOne: '#FFFFFF',
        colorTwo: '#FFFFFF',
        colorThree: '#FFFFFF',
        colorFour: '#FFFFFF',
        colorFive: '#FFFFFF'
      }
    ];
    mockGrabPalette = jest.fn();
    mockUpdatePaletteName = jest.fn();
    wrapper = shallow(<PaletteContainer 
      palettes={mockPalettes}
      grabPalette={mockGrabPalette}
      updatePaletteName={mockUpdatePaletteName}
    />);
  });

  it('should match the snapshot, given a set of colors', () => {
    expect(wrapper).toMatchSnapshot();
  });

});