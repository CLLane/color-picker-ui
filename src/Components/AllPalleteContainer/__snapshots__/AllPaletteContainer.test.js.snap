// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AllPalette Container should match the snapshot, given a set of colors 1`] = `
<section
  className="all-palette__container"
>
  <AllPaletteCard
    grabPalette={[MockFunction]}
    key="1"
    palette={
      Object {
        "colorFive": "#FFFFFF",
        "colorFour": "#FFFFFF",
        "colorOne": "#FFFFFF",
        "colorThree": "#FFFFFF",
        "colorTwo": "#FFFFFF",
        "id": 1,
        "name": "Cool Colors",
        "project_id": "12345",
      }
    }
  />
  <AllPaletteCard
    grabPalette={[MockFunction]}
    key="2"
    palette={
      Object {
        "colorFive": "#FFFFFF",
        "colorFour": "#FFFFFF",
        "colorOne": "#FFFFFF",
        "colorThree": "#FFFFFF",
        "colorTwo": "#FFFFFF",
        "id": 2,
        "name": "Warm Colors",
        "project_id": "12345",
      }
    }
  />
</section>
`;
